libs/sdk-ui-kit/src/Dialog/ShareDialog/test/ShareDialog.test.tsx

```diff
@@ -0,0 +1,244 @@
+                shareStatus: "private",
+            };
+
+            const wrapper = await createComponent({ onApply: onApply }, [], [], [userAccessGrantee]);
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove grantee from list and submit callback for empty grantees list status is private and isUnderStrictControl is true ", async () => {
+            const granteeSelector = getGranteeSelector(userAccessGrantee.user);
+            const onApply = jest.fn();
+            const expectedPayload: ISharingApplyPayload = {
+                granteesToAdd: [],
+                granteesToDelete: [
+                    {
+                        granteeRef: userAccessGrantee.user.ref,
+                    },
+                ],
+                isUnderStrictControl: false,
+                isLocked: false,
+                shareStatus: "private",
+            };
+
+            const wrapper = await createComponent({ onApply: onApply }, [], [], [userAccessGrantee]);
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove grantee from list and submit callback when group all present is public and isUnderStrictControl is false", async () => {
+            const granteeSelector = getGranteeSelector(userAccessGrantee.user);
+            const onApply = jest.fn();
+            const expectedPayload: ISharingApplyPayload = {
+                granteesToAdd: [],
+                granteesToDelete: [
+                    {
+                        granteeRef: userAccessGrantee.user.ref,
+                    },
+                ],
+                isUnderStrictControl: false,
+                isLocked: false,
+                shareStatus: "public",
+            };
+            const sharedObject: ISharedObject = { ...defaultSharedObject, shareStatus: "public" };
+            const wrapper = await createComponent(
+                { onApply: onApply, sharedObject },
+                [],
+                [],
+                [userAccessGrantee],
+            );
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove group all from list and submit callback is private", async () => {
```

Comment:
"should remove group all from list and submit callback with object set as private"

Reply:
renamed